pkg/database/realm.go

```diff
@@ -42,6 +48,19 @@ type Realm struct {
 	// Name is the name of the realm.
 	Name string `gorm:"type:varchar(200);unique_index"`
 
+	// Code configuration
+	RegionCode              string `gorm:"type:varchar(5); not null; default: ''"`
+	CodeLength              uint   `gorm:"type:smallint; not null; default: 8"`
+	CodeDurationSeconds     int64  `gorm:"type:bigint; not null; default: 900"` // default 15m (in seconds)
+	UseSMSLongCodes         bool   `gorm:"type:boolean; not null; default: true"`
+	LongCodeLength          uint   `gorm:"type:smallint; not null; default: 16"`
+	LongCodeDurationSeconds int64  `gorm:"type:bigint; not null; default: 86400"` // default 24h
+	// SMS Content
+	DeepLinkProtocol string `gorm:"type:varchar(50); not null; default: 'ens://v?'"`
+	DeepLinkRegion   bool   `gorm:"type:boolean; not null; default: true"`
```

Comment:
I've had a lot of problems with gorm and boolean types. This is another reason I vote for making an arbitrary template field the admin can provide.